### PR TITLE
prototype env vars

### LINKED ISSUE #206: Environment variables for testnet and mainnet urls

### PR DESCRIPTION
**Pull request by nguvictor**
Fixes #206

### CODE REVIEW

**Comment by EvgeniiaVak** [comment]
@nguvictor do you need help with fixing env vars for `.github/workflows/lint-and-test.yml` worflow for this PR?

**Comment by EvgeniiaVak** [comment]
can we use only `NETWORK=stokenet` in the env vars (without `NETWORK_ID=2`), we can then specify in the code which ID to use? (will be easier to test by changing only one variable

File: next.config.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ const nextConfig = {
   reactStrictMode: false,
 
   pageExtensions: ["js", "jsx", "mdx", "ts", "tsx"],
+  env: {
+    NETWORK: process.env.NETWORK,
+    DAPP_DEFINITION_ADDRESS: process.env.DAPP_DEFINITION_ADDRESS,
+  },
```
Comment (EvgeniiaVak) [inline next.config.js:16]: Do we really need this doubling? next should be able to pick up the env vars by default

File: src/app/subscriptions.ts [hunk 2]
```diff
@@ -48,7 +60,19 @@ export function initializeSubscriptions(store: AppStore) {
...
+      network = adex.ApiNetworkOptions.indexOf("mainnet");
+      break;
+    case "stokenet":
+      network = adex.ApiNetworkOptions.indexOf("stokenet");
+      break;
+    default:
+      network = adex.ApiNetworkOptions.indexOf("stokenet");
+  }
+
+  adex.init(adex.ApiNetworkOptions[network]);
```
Comment (EvgeniiaVak) [inline src/app/subscriptions.ts:75]:
```suggestion
  adex.init(process.env.NETWORK || "stokenet");
```

**Review by EvgeniiaVak — CHANGES_REQUESTED** [review]

File: src/app/subscriptions.ts (hunk 2 above)
Comment (nguvictor) [reply src/app/subscriptions.ts:75]: This was the first thing I tried, but ran into type issues.
The adex.init() is a bit strange it wants to have alphadexApiNetwork
image

image

File: next.config.js (hunk 1 above)
Comment (nguvictor) [reply next.config.js:16]: Ran into an issue that they were undefined in browser. I followed a Stackoverflow said that you had to do this for browser/client side. Looks like the new docs suggest another strategy.
https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser

Comment (EvgeniiaVak) [reply next.config.js:16]: maybe using `NEXT_PUBLIC_` prefix for the env vars?

**Review by EvgeniiaVak — APPROVED** [review]
I tried locally with mainnet:

```env
NEXT_PUBLIC_NETWORK=mainnet # Options: mainnet or stokenet
NEXT_PUBLIC_DAPP_DEFINITION_ADDRESS=account_rdx168qrzyngejus9nazhp7rw9z3qn2r7uk3ny89m5lwvl299ayv87vpn5
```

works fine 👍

**Pull request merged**